mfp_functions/fetchSingleDate.js

```diff
@@ -27,14 +27,45 @@ var fetchSingleDate = function(username, date, fields, callback){
     var cols = {};
 
     //find and set column numbers of nutrient fields
-    $('#food').find('thead').find('tr').find('td').each(function(index, element){
+    $('#food').find('thead').find('tr').find('td').each(function (index, element) {
       var $element = $(element);
       var fieldName = $element.text().toLowerCase();
-      if (fieldName === "sugars") { fieldName = "sugar"; } // fixes MFP nutrient field name inconsistency
-      if (fieldName === "cholest") { fieldName = "cholesterol"; } // fixes MFP nutrient field name inconsistency
-      if (index !== 0) { cols[fieldName] = index; } //ignore first field, which just says "Foods"
+      if (fieldName === "sugars") {
+        fieldName = "sugar";
+      } // fixes MFP nutrient field name inconsistency
+      if (fieldName === "cholest") {
+        fieldName = "cholesterol";
+      } // fixes MFP nutrient field name inconsistency
+      if (index !== 0) {
+        cols[fieldName] = index;
+      } //ignore first field, which just says "Foods"
     });
 
+    var lastMeal;
+    var meals = {
+      'breakfast': [],
+      'lunch': [],
+      'dinner': [],
+      'snacks': []
+    };
```

Comment:
these meals names are way too hard coded and specific.

The names can be arbitrary, and there can be an arbitrary number of them.